File: angular15-starter-project-main/src/app/services/ordreServ/ordre-serv.service.ts
import { HttpClient, HttpParams } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { BehaviorSubject, Observable } from 'rxjs';
import { environment } from 'src/environnement';

@Injectable({
  providedIn: 'root'
})
export class OrdreServService {
  private apiUrl = `${environment.apiUrl}/OrdreService`;
  private ordreServiceSubject = new BehaviorSubject<any[]>([]);
  ordreService$ = this.ordreServiceSubject.asObservable();

  constructor(private http: HttpClient) { }

  getOrdreServices(numMarche: number, numEtape: number) : Observable<any>{
    const params = new HttpParams()
    .set('numMarche', numMarche)
    .set('numEtape', numEtape);
    return this.http.get<any>(`${this.apiUrl}`, { params });
  }

  saveOrUpdateOrdreService(ordreServices: any[]): Observable<any[]> {
    return this.http.post<any[]>(`${this.apiUrl}`, ordreServices);
  }

  deleteOrdreService(numMarche: number, numEtape: number, numOs: number): Observable<any> {
    return this.http.delete<any>(`${this.apiUrl}/${numMarche}/${numEtape}/${numOs}`);
  }


}

File: src/main/java/com/afh/gescomp/service/OrdreServService.java
package com.afh.gescomp.service;

import com.afh.gescomp.model.primary.OrdreService;
import java.util.List;

public interface OrdreServService {
    List<OrdreService> getOrdreServiceByNumMarcheAndNumEtape(Long numMarche, Short numEtape);
    List<OrdreService> saveOrUpdateOrdreService(List<OrdreService> ordreServices);
    OrdreService findByNumMarcheAndNumEtapeAndNumOs(Long numMarche, Short numEtape, Integer numOs);
    void deleteOrdreService(Long numMarche, Short numEtape, Integer numOs);
}
